### PR TITLE
Rename of DNANucleotide, RNANucleotide, Nucleotide

### PR DESCRIPTION
**Pull request by Ward9250**
Implements renames as per #391.

### CODE REVIEW

File: docs/src/man/seq.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ ambiguous nucleotide can take. For example, `DNA_R` (meaning the nucleotide is
 either `DNA_A` or `DNA_G`) is encoded as `0101` because `0101` is the bitwise OR
 of `0001` (`DNA_A`) and `0100` (`DNA_G`). The gap symbol is always `0000`.
 
-|    Nucleotide    |  Bits  |
+|    NucleicAcid    |  Bits  |
```
Comment (kdmurray91) [inline docs/src/man/seq.md]: Extreme pedantry here: Is it worth bothering to line `|` chars up in these tables? If this is annoying in your editor, I can do it later using Tabular.vim

File: src/seq/deprecated.jl [hunk 2]
```diff
@@ -7,7 +7,7 @@ import Base.depwarn
 @deprecate ispalindrome ispalindromic
 
 """
-Range type of biological symbols (especially `DNANucleotide`, `RNANucleotide`,
+Range type of biological symbols (especially `DNA`, `RNA`,
```
Comment (kdmurray91) [inline src/seq/deprecated.jl]: This probably fits all on one line now?

File: docs/src/man/seq.md [hunk 3]
```diff
@@ -694,7 +694,7 @@ end
 ```
 
 
-## Nucleotide k-mers
+## NucleicAcid k-mers
```
Comment (bicycle1885) [inline docs/src/man/seq.md]: "Nucleic acid k-mers" may be better.

File: src/seq/composition.jl [hunk 4]
```diff
@@ -109,10 +109,10 @@ function Base.merge!{T}(comp::Composition{T}, other::Composition{T})
 end
 
 function Base.summary{T}(::Composition{T})
-    if T == DNANucleotide
-        return "DNA Nucleotide Composition"
-    elseif T == RNANucleotide
-        return "RNA Nucleotide Composition"
+    if T == DNA
+        return "DNA NucleicAcid Composition"
```
Comment (bicycle1885) [inline src/seq/composition.jl]: "DNA Composition" is better now?

File: src/seq/composition.jl [hunk 5]
```diff
@@ -109,10 +109,10 @@ function Base.merge!{T}(comp::Composition{T}, other::Composition{T})
...
+    elseif T == RNA
+        return "RNA NucleicAcid Composition"
```
Comment (bicycle1885) [inline src/seq/composition.jl]: "RNA Composition"?

File: src/seq/nucleotide.jl [hunk 6]
```diff
@@ -1,17 +1,17 @@
-# Nucleotides
+# NucleicAcids
```
Comment (bicycle1885) [inline src/seq/nucleotide.jl]: Rename the file name as well?

File: src/seq/nucleotide.jl [hunk 7]
```diff
@@ -167,27 +167,27 @@ true
...
-# DNA Nucleotides
+# DNA NucleicAcids
 
 # lookup table for characters
 const char_to_dna = [0x80 for _ in 0x00:0xff]
 const dna_to_char = Vector{Char}(16)
 
 # derived from "The DDBJ/ENA/GenBank Feature Table Definition"
-# §7.4.1 Nucleotide base code (IUPAC)
+# §7.4.1 NucleicAcid base code (IUPAC)
```
Comment (bicycle1885) [inline src/seq/nucleotide.jl]: This is the title of the section, so it should not be renamed.

File: src/seq/nucleotide.jl [hunk 8]
```diff
@@ -167,27 +167,27 @@ true
...
     return reinterpret(UInt8, nt)
 end
 
 
-# Nucleotide encoding definition
+# NucleicAcid encoding definition
 # ------------------------------
 
-# DNA Nucleotides
+# DNA NucleicAcids
```
Comment (bicycle1885) [inline src/seq/nucleotide.jl]: "NucleicAcids" may be redundant.

File: src/seq/nucleotide.jl [hunk 9]
```diff
@@ -208,19 +208,19 @@ for (char, doc, bits) in [
...
 
-@eval alphabet(::Type{DNANucleotide}) = $(tuple([reinterpret(DNANucleotide, x)
+@eval alphabet(::Type{DNA}) = $(tuple([reinterpret(DNA, x)
                                                  for x in 0b0000:0b1111]...))
 
 const ACGT = (DNA_A, DNA_C, DNA_G, DNA_T)
 const ACGTN = (DNA_A, DNA_C, DNA_G, DNA_T, DNA_N)
 
-# RNA Nucleotides
+# RNA NucleicAcids
```
Comment (bicycle1885) [inline src/seq/nucleotide.jl]: "NucleicAcids" may be redundant.

File: src/seq/nucleotide.jl [hunk 10]
```diff
@@ -261,63 +261,63 @@ const ACGUN = (RNA_A, RNA_C, RNA_G, RNA_U, RNA_N)
...
+function Base.show(io::IO, nt::DNA)
     if isvalid(nt)
         if nt == DNA_Gap
             write(io, "DNA_Gap")
         else
             write(io, "DNA_", Char(nt))
         end
     else
-        write(io, "Invalid DNA Nucleotide")
+        write(io, "Invalid DNA NucleicAcid")
```
Comment (bicycle1885) [inline src/seq/nucleotide.jl]: "Invalid DNA"

File: src/seq/nucleotide.jl [hunk 11]
```diff
@@ -261,63 +261,63 @@ const ACGUN = (RNA_A, RNA_C, RNA_G, RNA_U, RNA_N)
...
+function Base.show(io::IO, nt::RNA)
     if isvalid(nt)
         if nt == RNA_Gap
             write(io, "RNA_Gap")
         else
             write(io, "RNA_", Char(nt))
         end
     else
-        write(io, "Invalid RNA Nucleotide")
+        write(io, "Invalid RNA NucleicAcid")
```
Comment (bicycle1885) [inline src/seq/nucleotide.jl]: "Invalid RNA"

**Comment by Ward9250** [comment]
Addressed comments, thanks guys! I'll merge in the morning if there's nothing else to change.

**Pull request merged**